lib/flutter_map_tappable_polyline.dart

```diff
@@ -99,28 +101,60 @@ class TappablePolylineLayer extends StatelessWidget {
         return Container(
           child: GestureDetector(
               onTapUp: (TapUpDetails details) {
-                TaggedPolyline polyline = polylineOpts.polylines.firstWhere(
-                    (TaggedPolyline polylineOpt) => polylineOpt.offsets.firstWhere(
-                        (Offset offset) =>
-                            (offset.dx / polylineOpts.pointerDistanceTolerance)
-                                        .round()
-                                        .toDouble() *
-                                    polylineOpts.pointerDistanceTolerance ==
-                                (details.localPosition.dx / polylineOpts.pointerDistanceTolerance)
-                                        .round() *
-                                    polylineOpts.pointerDistanceTolerance &&
-                            (offset.dy / polylineOpts.pointerDistanceTolerance)
-                                        .round()
-                                        .toDouble() *
-                                    polylineOpts.pointerDistanceTolerance ==
-                                (details.localPosition.dy /
-                                            polylineOpts.pointerDistanceTolerance)
-                                        .round() *
-                                    polylineOpts.pointerDistanceTolerance,
-                        orElse: () => null) is Offset,
-                    orElse: () => null);
-
-                if (polyline is TaggedPolyline) onTap(polyline);
+                // Calculating taps in between points on the polyline. We
+                // iterate over all the segments in the polyline to find any
+                // matches with the tapped point within t he
+                // pointerDistanceTolerance.
+                polylineOpts.polylines.forEach((element) {
```

Comment:
Using `currentPolyline` here was great, could you please rename this variable to be clearer?